satpy/utils.py

```diff
@@ -261,3 +264,36 @@ def atmospheric_path_length_correction(data, cos_zen, limit=88.):
     corr = corr.where(cos_zen > limit).fillna(corr_lim)
 
     return data * corr
+
+
+def scn_to_xrds(scn, datasets = None):
```

Comment:
In addition to the stickler suggested fix, could this be used to resolve #367? @mraspaud what were your thoughts on that? A `Scene.to_dataset` method might be too confusing.

Either way I think I'd like this utility function to be called `scene_to_xarray_dataset` instead of the abbreviations.

Reply:
Yes renaming is fine and makes things more clear. I already saw the issue you opened and really like that idea. Also talked to @mraspaud about that in Darmstadt.
It brought us to the point that it woud be nice if the DatasetID, which is limited to the values defined in the named tuple right now, could be more dynamic (that would open up more possibilities e.g. if two platforms with the same channels are in a Scene/MultiScene they should be selectable by other information). Ideally the DatasetID would (also) use the information of the xr.Dataset attributes.

I think @mraspaud had an idea about the dynamic DatasetID in Darmstadt and already started trying something out.

One more thing:
How do I go about your suggested changes? Should I correct them and do another PR or can I somehow edit the PR right here (have not really worked with the github interface up to now).

Reply:
Yes, we need to work on the datasetID stuff, but it's not needed for this function to work. Look also at what is done in the cf writer, as we transform the scene to a xr.Dataset instance prior to saving to netcdf.

Reply:
My not-so-secret "agenda" is to merge this before the 0.11 release later this month if at all possible. Or at least by early January because I'd like to show some examples in a SatPy talk I have at the AMS Meeting in Phoenix the second week of January.

I think the function is big enough that it doesn't need to be moved to the `Scene` but maybe a `to_xarray_dataset` method could be added to the Scene that imports and calls this utility function? Or is that too redundant/roundabout?

Reply:
Oh and @BENR0, you can push any new commits to this same branch on your fork and the PR will be automatically updated with the changes. Typically github realizes that you changed the code for a particular discussion and will mark the conversation as "resolved" or "outdated".

Reply:
No you are right just a normal imstande method.